src/util/logger.rs

```diff
@@ -0,0 +1,389 @@
+// Pruned copy of crate rust log, without global logger
+// https://github.com/rust-lang-nursery/log #7a60286
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::cmp;
+use std::fmt;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use std::ascii::AsciiExt;
+
+static LOG_LEVEL_NAMES: [&'static str; 6] = ["OFF", "ERROR", "WARN", "INFO", "DEBUG", "TRACE"];
+
+static LEVEL_PARSE_ERROR: &'static str =
+		"attempted to convert a string that doesn't match an existing log level";
+
+/// An enum representing the available verbosity levels of the logger.
+#[derive(Copy, Eq, Debug, Hash)]
+pub enum Level {
+	///Designates logger being silent
+	Off,
+	/// Designates very serious errors
+	Error = 1,
+	/// Designates hazardous situations
+	Warn,
+	/// Designates useful information
+	Info,
+	/// Designates lower priority information
+	Debug,
+	/// Designates very low priority, often extremely verbose, information
+	Trace,
+}
+
+impl Clone for Level {
+	#[inline]
+	fn clone(&self) -> Level {
+		*self
+	}
+}
+
+impl PartialEq for Level {
+	#[inline]
+	fn eq(&self, other: &Level) -> bool {
+		*self as usize == *other as usize
+	}
+}
+
+impl PartialOrd for Level {
+	#[inline]
+	fn partial_cmp(&self, other: &Level) -> Option<cmp::Ordering> {
+		Some(self.cmp(other))
+	}
+
+	#[inline]
+	fn lt(&self, other: &Level) -> bool {
+		(*self as usize) < *other as usize
+	}
+
+	#[inline]
+	fn le(&self, other: &Level) -> bool {
+		*self as usize <= *other as usize
+	}
+
+	#[inline]
+	fn gt(&self, other: &Level) -> bool {
+		*self as usize > *other as usize
+	}
+
+	#[inline]
+	fn ge(&self, other: &Level) -> bool {
+		*self as usize >= *other as usize
+	}
+}
+
+impl Ord for Level {
+	#[inline]
+	fn cmp(&self, other: &Level) -> cmp::Ordering {
+		(*self as usize).cmp(&(*other as usize))
+	}
+}
+
+fn ok_or<T, E>(t: Option<T>, e: E) -> Result<T, E> {
+	match t {
+		Some(t) => Ok(t),
+		None => Err(e),
+	}
+}
+
+impl FromStr for Level {
+	type Err = ParseLevelError;
+	fn from_str(level: &str) -> Result<Level, Self::Err> {
+		ok_or(
+			LOG_LEVEL_NAMES
+				.iter()
+				.position(|&name| str::eq_ignore_ascii_case(name, level))
+				.into_iter()
+				.filter(|&idx| idx != 0)
+				.map(|idx| Level::from_usize(idx).unwrap())
+				.next(),
+			ParseLevelError(()),
+		)
+	}
+}
+
+impl fmt::Display for Level {
+	fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+		fmt.pad(LOG_LEVEL_NAMES[*self as usize])
+	}
+}
+
+#[allow(dead_code)]
```

Comment:
Can you just add a few sample logs in obvious places and remove the allow(dead_code)s? That way we can see it works and no need for bending over backwards here.